### ISSUE TITLE
[BUG] [C-libcurl] If the name of a function parameter gets escaped for C keyword,  the other parts in function must use the escaped parameter name

### ISSUE THREAD

**Issue by ityuhui** [OpenAPITools/openapi-generator#4782]
#### Bug Report Checklist

- [X] Have you provided a full/minimal spec to reproduce the issue?
- [X] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [X] What's the version of OpenAPI Generator used?
- [X] Have you search for related issues/PRs?
- [X] What's the actual output vs expected output?
- [ ] [Optional] Bounty to sponsor the fix ([example](https://www.bountysource.com/issues/66123212-javascript-client-produces-a-wrong-object-for-a-string-enum-type-that-is-used-with-ref))

##### Description

When using C-libcurl generator to generate kubernetes API, some parameter names of function are keywords of C programming language, e.g. **continue**

```
AppsV1API_deleteAppsV1CollectionNamespacedControllerRevision(char *continue
...
if (continue)
    {
        keyQuery_continue = strdup("continue");
        valueQuery_continue = strdup((continue));

```

They will cause building error.
I try to fix this issue by below diff

```
--- a/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CLibcurlClientCodegen.java
+++ b/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CLibcurlClientCodegen.java
@@ -337,7 +337,7 @@ public class CLibcurlClientCodegen extends DefaultCodegen implements CodegenConf
     @Override
     public String toParamName(String name) {
         // should be the same as variable name
-        if (name.matches("^\\d.*")) {
+        if (isReservedWord(name) || name.matches("^\\d.*")) {
             name = escapeReservedWord(name);
         }
         name = name.replaceAll("-","_");
```

And then the parameter names are OK, but other parts must use the escaped value:

```
AppsV1API_deleteAppsV1CollectionNamespacedControllerRevision(char *_continue
...
if (_continue)
    {
        keyQuery__continue = strdup("_continue");
        valueQuery__continue = strdup((_continue));

```

keyQuery__continue = strdup("_continue"); is not right, it may cause logic error.

I found this code is generated by below template in [api-body.mustache](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/C-libcurl/api-body.mustache):

```
if ({{paramName}})
    {
        ...
        keyQuery_{{{paramName}}} = strdup("{{{paramName}}}");
``` 

Does openapi-generator provide a variable for no-need-escaped-value for a param name ?
e.g. originalParamName

```
keyQuery_{{{paramName}}} = strdup("{{{originalParamName}}}");
```
##### openapi-generator version
master

##### OpenAPI declaration file content or url
https://raw.githubusercontent.com/kubernetes/kubernetes/master/api/openapi-spec/swagger.json

##### Command line used for generation
mvn clean package
java -jar modules/openapi-generator-cli/target/openapi-generator-cli.jar generate -i https://raw.githubusercontent.com/kubernetes/kubernetes/master/api/openapi-spec/swagger.json -g c -o ../c_k8s_api_client

##### Steps to reproduce
generate the c api client by the above command
check the file "c_k8s_api_client/api/AppsV1API.c"

##### Related issues/PRs
None

##### Suggest a fix
Does openapi-generator provide a variable for no-need-escaped-value for a param name ?

**Comment by wing328** [comment]
> Does openapi-generator provide a variable for no-need-escaped-value for a param name ?
e.g. originalParamName

Try `{{baseName}}`

**Comment by ityuhui** [comment]
Thank you @wing328 

I will submit a PR for this issue.

**Comment by ityuhui** [comment]
This issue is fixed by the [PR #4838](https://github.com/OpenAPITools/openapi-generator/pull/4838)
